airflow/contrib/hooks/gcp_transfer_hook.py

```diff
@@ -56,52 +85,282 @@ def get_conn(self):
+    def get_transfer_operation(self, operation_name):
+        """
+        Gets an transfer operation in Google Storage Transfer Service.
+
+        :param operation_name: Name of the transfer operation.
+        :type operation_name: str
+        :return: transfer operation
+        :rtype: dict
+        """
+        return self.get_conn()\
+            .transferOperations()\
+            .get(name=operation_name)\
+            .execute(num_retries=NUM_RETRIES)
+
+    @GoogleCloudBaseHook.convert_http_exception
+    def list_transfer_operations(self, filter):
+        """
+        Gets an transfer operation in Google Storage Transfer Service.
+
+        :param filter: A request filter, as described in
+            https://cloud.google.com/storage-transfer/docs/reference/rest/v1/transferJobs/list#body.QUERY_PARAMETERS.filter
+            With one additional improvement:
+
+            * project_id is optional if you have a project id defined
+              in the connection
+              See: :ref:`connection-type-GCP`
+
+        :type filter: dict
+        :return: transfer operation
+        :rtype: list[dict]
+        """
+        conn = self.get_conn()
+
+        filter = self._append_project_id(filter, 'filter project_id parameter')
+
+        operations = []
+
+        request = conn.transferOperations().list(
+            name=TRANSFER_OPERATIONS,
+            filter=json.dumps(filter)
+        )
+
+        while request is not None:
+            response = request.execute(num_retries=NUM_RETRIES)
+            if 'operations' in response:
+                operations.extend(response['operations'])
+
+            request = conn.transferOperations().list_next(
+                previous_request=request,
+                previous_response=response)
+
+        return operations
+
+    @GoogleCloudBaseHook.convert_http_exception
+    def pause_transfer_operation(self, operation_name):
+        """
+        Pauses an transfer operation in Google Storage Transfer Service.
+
+        :param operation_name: Name of the transfer operation.
+        :type operation_name: str
+        :rtype: None
+        """
+        self.get_conn()\
+            .transferOperations()\
+            .pause(name=operation_name)\
+            .execute(num_retries=NUM_RETRIES)
+
+    @GoogleCloudBaseHook.convert_http_exception
+    def resume_transfer_operation(self, operation_name):
+        """
+        Resumes an transfer operation in Google Storage Transfer Service.
+
+        :param operation_name: Name of the transfer operation.
+        :type operation_name: str
+        :rtype: None
+        """
+        self.get_conn()\
+            .transferOperations()\
+            .resume(name=operation_name)\
+            .execute(num_retries=NUM_RETRIES)
+
+    @GoogleCloudBaseHook.convert_http_exception
+    def wait_for_transfer_job(self,
+                              job,
+                              expected_status=[GcpTransferOperationStatus.SUCCESS]):
```

Comment:
Should be tuple (GcpTransferOperationStatus.SUCCESS,) - the default parameter should not be mutable, otherwise you can modify it inside the method call and get it changed for another method call.

Reply:
@potiuk Good point. I did not notice this.